dot/state/storage.go

```diff
@@ -321,6 +321,19 @@ func (s *StorageState) Entries(hash *common.Hash) (map[string][]byte, error) {
 	return s.tries[*hash].Entries(), nil
 }
 
+// GetKeysWithPrefix returns all that match the given prefix for the given hash (or best block state root if hash is nil) in lexicographic order
+func (s *StorageState) GetKeysWithPrefix(hash *common.Hash, prefix []byte) ([][]byte, error) {
+	if hash == nil {
+		sr, err := s.blockState.BestBlockStateRoot()
+		if err != nil {
+			return nil, err
+		}
+		hash = &sr
+	}
+
+	return s.tries[*hash].GetKeysWithPrefix(prefix), nil
```

Comment:
can you error or load trie from database in the case `s.tries[*hash]` is nil?

Reply:
yes, good catch.  All set.